DUnitX.Assert.pas

```diff
@@ -48,21 +49,22 @@ interface
   Assert = class
   private
     class var fOnAssert: TProc;
-    class var fTestFailure: ExceptClass;
```

Comment:
In that case you might want to add your new feature either as helper for the Assert class or via Event that can produce the proper exception. In any case try to keep Assert as simple as possible without cluttering it with special framework related cases (personally I see the diffing logic as special case).

I for example also had the problem of comparing long strings (xml) via unit test and the DUnit error message was not very helpful telling me that <thousands-of-lines-xml-string-1> differs from <thousands-of-lines-xml-string-2> so I implemented my own [CheckEqualsString](https://bitbucket.org/sglienke/spring4d/src/fad135e1c6000d48fa12aff747274193e7439cb8/Tests/Source/Spring.TestUtils.pas?at=release%2F1.2&fileviewer=file-view-default#Spring.TestUtils.pas-95) method that only shows a snippet where the strings differ and the position where that is)